pkg/core/auth/session/writer_impl.go

```diff
@@ -56,20 +57,38 @@ func (w *writerImpl) WriteSession(rw http.ResponseWriter, accessToken *string) {
 		cookie.Expires = time.Unix(0, 0)
 	}
 
-	http.SetCookie(rw, cookie)
+	updateCookie(rw, cookie)
 }
 
 func (w *writerImpl) ClearSession(rw http.ResponseWriter) {
-	clientConfig := w.clientConfigs[w.authContext.AccessKey().ClientID]
-	useCookie := clientConfig.SessionTransport == config.SessionTransportTypeCookie
-	if useCookie {
-		cookie := &http.Cookie{
-			Name:     coreHttp.CookieNameSession,
-			Path:     "/",
-			HttpOnly: true,
-			Secure:   !w.useInsecureCookie,
-			Expires:  time.Unix(0, 0),
+	cookie := &http.Cookie{
+		Name:     coreHttp.CookieNameSession,
+		Path:     "/",
+		HttpOnly: true,
+		Secure:   !w.useInsecureCookie,
+		Expires:  time.Unix(0, 0),
+	}
+	updateCookie(rw, cookie)
+}
+
+func updateCookie(rw http.ResponseWriter, cookie *http.Cookie) {
+	headers := rw.Header()
```

Comment:
Can we just ask the stdlib to parse for us? https://golang.org/pkg/net/http/#Response.Cookies

The idea is to call Response.Cookies to get parsed cookies. Update the target. Remove all `Set-Cookie` headers from the map. Set the updated cookies with `http.SetCookie` again. The advantage is we do not need to parse the serialized cookie header value ourselves.

Reply:
I had considered this before, but it feels quite hacky to me: we'll need to make a dummy `http.Response` with the header set to use its `Cookies` method. If you feel that it is better than a manual parsing (which is extracted from stdlib), I'll change it to use that way.

Reply:
Let's create a dummy Response and call its Cookies method